pandas/core/generic.py

```diff
@@ -5906,6 +5906,95 @@ def infer_objects(self: FrameOrSeries) -> FrameOrSeries:
             )
         ).__finalize__(self)
 
+    def as_nullable_dtypes(
+        self: FrameOrSeries, keep_integer: bool_t = False
+    ) -> FrameOrSeries:
+        """
+        Convert columns of DataFrame or a Series to types supporting ``pd.NA``.
+
+        If the dtype is "object", convert to "string", "boolean" or an appropriate
+        integer type.
```

Comment:
what about objects that are not strings?

Reply:
clarified.